File: 8-semester-pamo/exercise-3/Tipper/app/src/main/java/com/example/tipper/QuizResult.java
package com.example.tipper;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class QuizResult extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.quiz_result);

        TextView resultsTextView = findViewById(R.id.resultsTextView);

        int correctAnswers = getIntent().getIntExtra("correctAnswers", 0);
        int totalQuestions = getIntent().getIntExtra("totalQuestions", 0);

        resultsTextView.setText("You got " + correctAnswers + " out of " + totalQuestions + " correct!");

        Button returnButton = findViewById(R.id.returnButton);
        returnButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(QuizResult.this, Menu.class);
                startActivity(intent);
            }
        });

        findViewById(R.id.endQuizButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
}

File: 8-semester-pamo/exercise-4/Tipper/app/src/main/java/com/example/tipper/QuizResult.kt
package com.example.tipper

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class QuizResult : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.quiz_result)
        val resultsTextView = findViewById<TextView>(R.id.resultsTextView)
        val correctAnswers = intent.getIntExtra("correctAnswers", 0)
        val totalQuestions = intent.getIntExtra("totalQuestions", 0)
        resultsTextView.text = "You got $correctAnswers out of $totalQuestions correct!"
        val returnButton = findViewById<Button>(R.id.returnButton)
        returnButton.setOnClickListener {
            val intent = Intent(this@QuizResult, Menu::class.java)
            startActivity(intent)
        }
        findViewById<View>(R.id.endQuizButton).setOnClickListener { finish() }
    }
}